flaky/flaky_pytest_plugin.py

```diff
@@ -84,12 +32,152 @@ class FlakyPlugin(_FlakyPlugin):
+        :type item:
+            :class:`Function`
+        :param nextitem:
+            py.test wrapper for the next test function to be run
+        :type nextitem:
+            :class:`Function`
+        """
+        test_instance = self._get_test_instance(item)
+        self._copy_flaky_attributes(item, test_instance)
+        if self.force_flaky and not self._has_flaky_attributes(item):
+            self._make_test_flaky(
+                item,
+                self.max_runs,
+                self.min_passes,
+            )
+        patched_call_runtest_hook = self.runner.call_runtest_hook
+        self._call_infos = {}
+        should_rerun = True
+        try:
+            self.runner.call_runtest_hook = self.call_runtest_hook
+            while should_rerun:
+                self.runner.pytest_runtest_protocol(item, nextitem)
+                call_info = self._call_infos.get('call', None)
+                if call_info is None:
+                    return
+                run = self._call_infos['call']
+                passed = run.excinfo is None
+                if passed:
+                    should_rerun = self.add_success(item)
+                else:
+                    should_rerun = self.add_failure(item, run.excinfo)
+                    if not should_rerun:
+                        item.excinfo = run.excinfo
+        finally:
+            self.runner.call_runtest_hook = patched_call_runtest_hook
+            del self._call_infos
+        return True
+
+    @pytest.hookimpl(tryfirst=True, hookwrapper=True)
+    def pytest_runtest_makereport(self, item, call):
+        """
+        Pytest hook to intercept the report for reruns.
+
+        Change the report's outcome to 'passed' if flaky is going to handle the test run.
+        That way, pytest will not mark the run as failed.
+        """
+        outcome = yield
+        if call.when == 'call':
+            report = outcome.get_result()
+            report.item = item
+            report.original_outcome = report.outcome
+            if report.failed:
+                if self._should_handle_test_error_or_failure(item, None, None)[1]:
+                    report.outcome = 'passed'
+
+    @pytest.hookimpl(tryfirst=True, hookwrapper=True)
+    def pytest_report_teststatus(self, report):
+        """
+        Pytest hook to only add final runs to the report.
+
+        Given a test report, get the correpsonding test status.
+        For tests that flaky is handling, return the empty status
+        so it isn't reported; otherwise, don't change the status.
+        """
+        outcome = yield
+        if report.when == 'call':
+            item = report.item
+            if report.original_outcome == 'passed':
+                if self._should_handle_test_success(item):
+                    outcome.force_result(('', '', ''))
+            elif report.original_outcome == 'failed':
+                if self._should_handle_test_error_or_failure(item, None, None)[1]:
```

Comment:
I think I also made this comment very far above, but maybe a namedtuple for the return value of this method. [1] isn't clear.